src/controllers/trainee/Controller.ts

```diff
@@ -0,0 +1,85 @@
+import { Request, Response } from 'express';
+import * as bcrypt from 'bcrypt';
+import UserRepository from '../../repositories/user/UserRepository';
+import SystemResponse from '../../libs/SystemResponse';
+class TraineeController {
+    static instance;
+    private userRepository: UserRepository = new UserRepository();
+    static getInstance() {
+        if (TraineeController.instance)
+         return TraineeController;
+        TraineeController.instance = new TraineeController();
+          return TraineeController.instance;
+    }
+    create = async (req: Request, res: Response) => {
+        console.log('::::CREATE TRAINEE:::::');
+        try {
+            const traineeData = req.body;
+            const hash = await bcrypt.hash(traineeData.password, 10);
+            const trainee = await this.userRepository.create({ ...traineeData, password: hash});
+                return SystemResponse.success(res, trainee, 'Trainee Added Successfully');
+        }
+        catch (error) {
+                return SystemResponse.error(res, error, 'Trainee Added UnSuccessfull');
+            }
+    }
+    update = async (req: Request, res: Response) => {
+        console.log('::::UPDATE TRAINEE:::::');
+        try {
+            const traineeData = req.body;
+            const traineeId = traineeData.id;
+        const trainee = await this.userRepository.update1(traineeData.id, traineeData.dataToUpdate, traineeId);
+                return SystemResponse.success(res, trainee, 'Trainee Updated Successfully');
+        }
+        catch (error) {
+                return SystemResponse.error(res, error, 'Trainee Update UnSuccessfull');
+            }
+    }
+    list = async (req: Request, res: Response) => {
+        console.log(':::TRAINEE LIST:::::');
+        try {
+            let trainee;
+            let sort: any;
+            if (req.query.sort === 'email') {
+                sort = {email: 1 };
+            }
+            else if (req.query.sort === 'name') {
+                sort = {name: 1 };
+            }
+            else
+            sort = { updatedAt: 1 };
+            let search: any;
+            if (req.query.searchBy !== undefined) {
```

Comment:
Why are we calling list method inside conditions?

Don't you think we can simply make an object of query and pass it on to list method and everything will be fine